sdk/inc/azure/iot/az_iot_hub_client.h

```diff
@@ -64,11 +63,10 @@ typedef struct
   az_span user_agent;
 
   /**
-   * The `method_twin_content_type` is sent in the username to inform IoT Hub what format the device
-   * expects Twin Document and Direct Method payloads to use. If this option is not set, the default
-   * is JSON.
+   * The `twin_content_type` is sent in the username to inform IoT Hub what format the device
+   * expects Twin Document to use. If this option is not set, the default is JSON.
    */
-  az_span method_twin_content_type;
+  az_iot_hub_client_option_values twin_content_type;
```

Comment:
can we put this at the end of the struct? That way new options are always added at the end.

Reply:
The reason I put it here is to follow the logic order in the building of the username. model_id has to go at the end.  It keeps the order consistent for the reader.

Reply:
CC @ahsonkhan do we have guidance on where to put new options? I'm thinking from a compatibility standpoint leaving already released options in the same point in the struct would be beneficial and any new options should be tacked on to the end?

Reply:
I'm curious to know more.  What kind of compatibility issues might arise here?  It's not an enum, so element number assignments aren't being changed.  The only think I can see being affected by inserting this new element instead of tacking on, is that because it is of a different type than az_span, possibly the size of the struct is not optimized?  But i'm not sure I see a compatibility change.

Reply:
We don't ship dynamic library, so not an issue. But it is still good practice to follow.